src/ophys_etl/modules/denoising/fine_tuning/utils/data_splitter.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+from typing import Union, Tuple, List, Optional
+
+import h5py
+import numpy as np
+
+
+class DataSplitter:
+    """Splits movie into train/test sets of frames"""
+    def __init__(
+        self,
+        movie_path: Union[str, Path],
+        seed: Optional[int] = None
+    ):
+        """
+
+        Parameters
+        ----------
+        movie_path
+            Path to movie
+        seed
+            Seed for reproducibility
+        """
+        self._movie_path = Path(movie_path)
+        self._seed = seed
+
+    def get_train_val_split(
+            self,
+            train_frac: float = 0.7,
+            window_size: int = 30
+    ) -> Tuple[List[int], List[int]]:
+        """
+        Returns train/val split for the movie. Each split contains the center
+        frame indices.
+
+        Parameters
+        ----------
+        train_frac
+            The fraction of center frames to reserve for the training set
+        window_size
+            Number of frames before and after a center frame
+
+        Returns
+        -------
+        Tuple of train, val center frame indices
+        """
+        with h5py.File(self._movie_path, 'r') as f:
+            nframes = f['data'].shape[0]
+
+        # min, max center frame must be window_size away from beginning/end
+        # since the frame needs to have window_size before/after it
+        all_frames = np.arange(window_size,
```

Comment:
You clip the window_size from the end and beginning of the movie but not on the seam between the train and validation set. Any reason to not do this?

Reply:
You are right, there will be on training example that includes frames that overlap with an example in the validation set. Is this what you mean?

Reply:
Ah, I was thinking that they would still be separate but just run into the "wall" created by the split. I'd say it's worth it to be able to say these are fully distinct train/val splits. While the 30 frames is an edge case, better to get rid of it now and add a comment explaining the code.

Reply:
Let me clarify. The edge case you are thinking of is let's say the movie is 1060 frames long. 70% train, 30% val. The first 30 and last 30 frames get cut off. So left with 1000 frames. From this, the last train input is center frame 700. That means the input will extend from 670 - 730 if using a 60 frame window size. The first val center frame will be frame 701. It extends from 671 - 731. But once we get to frame 761 it won't overlap with train anymore. So 60 examples here would overlap with the train set. You are asking to make sure that any examples in the validation set don't overlap with frames from the train set?

Reply:
Thanks for clarifying. Yeah, I think that's what I trying to get at. My initial worry was that there was an edge that wasn't accounted for but, as you say, the real thing here is that there are 60 (i misspoke and said 30) frames that overlap between test and train. I was say remove the overlap (just so we are doing exactly what we say we are) and add a comment. Or you can keep it as is and add a comment clarifying the above that due to the 60 frame overlap, the train/val samples are technically full separate.